pkg/fakedri/fakedri.go

```diff
@@ -0,0 +1,497 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//---------------------------------------------------------------
+// sysfs SPECIFICATION
+//
+// sys/class/drm/cardX/
+// sys/class/drm/cardX/lmem_total_bytes (gpu memory size, number)
+// sys/class/drm/cardX/device/
+// sys/class/drm/cardX/device/vendor (0x8086)
+// sys/class/drm/cardX/device/sriov_numvfs (PF only, number of VF GPUs, number)
+// sys/class/drm/cardX/device/drm/
+// sys/class/drm/cardX/device/drm/cardX/
+// sys/class/drm/cardX/device/drm/renderD1XX/
+// sys/class/drm/cardX/device/numa_node (Numa node index[1], number)
+// [1] indexing these: /sys/devices/system/node/nodeX/
+//---------------------------------------------------------------
+// devfs SPECIFICATION
+//
+// dev/dri/cardX
+// dev/dri/renderD1XX
+//---------------------------------------------------------------
+
+package fakedri
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	dirMode         = 0775
+	fileMode        = 0644
+	cardBase        = 0
+	renderBase      = 128
+	maxDevs         = 128
+	sysfsPath       = "/tmp/sys"
+	devfsPath       = "/tmp/dev"
+	mib             = 1024.0 * 1024.0
+	devNullMajor    = 1
+	devNullMinor    = 3
+	devNullType     = unix.S_IFCHR
+	maxK8sLabelSize = 63
+	fullyConnected  = "FULL"
+)
+
+// genOptionsWithTags represents the struct for our YAML data.
+type GenOptions struct {
+	Capabilities  map[string]string `yaml:"Capabilities"` // Device capabilities mapping for NFD hook
+	Info          string            `yaml:"Info"`         // Verbal config description
+	Driver        string            `yaml:"Driver"`       // Driver name (i915, xe)
+	Mode          string            `yaml:"Mode"`         // Mode of operation (future use with different generation modes)
+	Path          string            `yaml:"Path"`         // Path to fake device folder
+	NfdFeatureDir string            `yaml:"NfdDir"`       // NFD directory
+	DevCount      int               `yaml:"DevCount"`     // How many devices to fake
+	TilesPerDev   int               `yaml:"TilesPerDev"`  // Per-device tile count
+	DevMemSize    int               `yaml:"DevMemSize"`   // Available per-device device-local memory, in bytes
+	DevsPerNode   int               `yaml:"DevsPerNode"`  // How many devices per Numa node
```

Comment:
Maybe 'DevsPerNumaNode' ? 'DevsPerNode' could also mean devices per node/host.

Reply:
Agreed i was thinking that in one point